datasets/flwr_datasets/partitioner/continuous_partitioner.py

```diff
@@ -0,0 +1,165 @@
+        seed: Optional[int] = 42,
+    ) -> None:
+        super().__init__()
+        if not 0 <= strictness <= 1:
+            raise ValueError("`strictness` must be between 0 and 1")
+        if num_partitions <= 0:
+            raise ValueError("`num_partitions` must be greater than 0")
+
+        self._num_partitions = num_partitions
+        self._partition_by = partition_by
+        self._strictness = strictness
+        self._shuffle = shuffle
+        self._seed = seed
+        self._rng = np.random.default_rng(seed)
+
+        # Lazy initialization
+        self._partition_id_to_indices: dict[int, list[int]] = {}
+        self._partition_id_to_indices_determined = False
+
+    def load_partition(self, partition_id: int) -> Dataset:
+        """Load a single partition based on the partition index.
+
+        Parameters
+        ----------
+        partition_id : int
+            The index that corresponds to the requested partition.
+
+        Returns
+        -------
+        dataset_partition : Dataset
+            A single dataset partition.
+        """
+        self._check_and_generate_partitions_if_needed()
+        return self.dataset.select(self._partition_id_to_indices[partition_id])
+
+    @property
+    def num_partitions(self) -> int:
+        """Total number of partitions."""
+        self._check_and_generate_partitions_if_needed()
+        return self._num_partitions
+
+    @property
+    def partition_id_to_indices(self) -> dict[int, list[int]]:
+        """Mapping from partition ID to dataset indices."""
+        self._check_and_generate_partitions_if_needed()
+        return self._partition_id_to_indices
+
+    def _check_and_generate_partitions_if_needed(self) -> None:
+        """Lazy evaluation of the partitioning logic."""
+        if self._partition_id_to_indices_determined:
+            return
+
+        if self._num_partitions > self.dataset.num_rows:
+            raise ValueError(
+                "Number of partitions must be less than or equal to number of dataset samples."
+            )
+
+        # Extract property values
+        property_values = np.array(self.dataset[self._partition_by], dtype=np.float32)
+
+        # Check for missing values (None or NaN)
+        if np.any(property_values is None) or np.isnan(property_values).any():
+            raise ValueError(
+                f"The column '{self._partition_by}' contains None or NaN values, "
+                "which are not supported by ContinuousPartitioner. "
+                "Please clean or filter your dataset before partitioning."
+            )
+
+        # Standardize
+        std = np.std(property_values)
+        if std < 1e-6:
+            raise ValueError(
+                f"Cannot standardize column '{self._partition_by}' because it has near-zero s.d. "
+                f"(std={std}). All values are nearly identical, which prevents meaningful"
+                " partitioning. Please choose a different column or adjust the dataset."
```

Comment:
Should this `ValueError` be always raised if std is very small? or only depending on the `strictness` value? For example, if `strictness` is 0 (so IID) we should always be able to partition it, right?